overlord/snapstate/snapstate.go

```diff
@@ -2978,7 +2978,7 @@ func SwitchToNewGadget(st *state.State, name string) (*state.TaskSet, error) {
 
 // AddGadgetAssetsTasks creates the same tasks as SwitchToNewGadget but adds
 // them to the provided task set.
-func AddGadgetAssetsTasks(st *state.State, ts *state.TaskSet) (*state.TaskSet, error) {
+func AddGadgetAssetsTasks(st *state.State, ts *state.TaskSet, fromChange string) (*state.TaskSet, error) {
```

Comment:
it's interesting that this function had no general conflict check, I wonder if because we make remodel themselves conflict with everything else, see code in conflict.go? Maybe we need less conflict checking but it's fair to explore this more later?

Reply:
Looking at `remodelEssentialSnapTasks`, it looks like the check from the kernel command line can be removed from there as it is done already by `snapstate.SwitchToNewGadget`, which will be called a bit later in that function. I've removed it now.